pkgs/shells/fish/default.nix

```diff
@@ -70,14 +70,21 @@ stdenv.mkDerivation rec {
     sed -i "s|command manpath|command ${man-db}/bin/manpath|"     \
             "$out/share/fish/functions/man.fish"
   '' + ''
-    tee -a $out/share/fish/config.fish << EOF
-
-    # make fish pick up completions from nix profile
-    if status --is-interactive
-      set -l profiles (echo \$NIX_PROFILES | ${coreutils}/bin/tr ' ' '\n')
-      set fish_complete_path \$profiles"/share/fish/vendor_completions.d" \$fish_complete_path
-    end
+#    tee -a $out/share/fish/config.fish << EOF
+#    # make fish pick up completions from nix profile
+#    if status --is-interactive
+#      set -l profiles (echo \$NIX_PROFILES | ${coreutils}/bin/tr ' ' '\n')[-1..1]
+#      set fish_complete_path \$profiles"/share/fish/vendor_completions.d" \$fish_complete_path
+#    end
+#    EOF
+  '' + ''
+    tee -a $out/share/fish/preinit-env.fish << EOF
+    # source NixOS preinit-env
+    test -e /etc/fish/preinit-env.fish
+    and source /etc/fish/preinit-env.fish
     EOF
+  '' + ''
+    tee -a $out/share/fish/__fish_build_paths.fish < ${./__fish_build_paths.suffix.fish}
```

Comment:
Because to do the same thing with cat one would need to explicitly use temporary files (you can't use cat to read from a file and modify the same file in-place), I think. I went with it because it was the method in place when I started working on this code. I'm aiming to replace it with a better way of handling fish configurations via Nix, but I think that work is properly out of scope for this PR. (See pending comment in the main thread for this PR)